website/project/model.py

```diff
@@ -511,6 +522,10 @@ def validate_user(value):
             raise ValidationValueError('User does not exist.')
     return True
 
+class NodeUpdateError(Exception):
+    def __init__(self, key, reason):
```

Comment:
To make the API more consistent with built-in exceptions, make `message` the first parameter here.

Also call the superclass's `__init__`

```python
class NodeUpdateError(Exception):
    def __init__(self, message, key):
        self.key = key
        super(NodeUpdateError, self).__init__(message)
```